File: source/android/app/src/main/java/com/qppd/ricedryer/data/repository/AuthRepository.java
package com.qppd.ricedryer.data.repository;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseUser;
import com.qppd.ricedryer.data.model.User;
import com.qppd.ricedryer.data.remote.FirebaseAuthManager;
import com.qppd.ricedryer.data.remote.FirebaseDataSource;

public class AuthRepository {
    private static AuthRepository instance;
    private final FirebaseAuthManager authManager;
    private final FirebaseDataSource dataSource;
    private final MutableLiveData<FirebaseUser> userLiveData;

    private AuthRepository() {
        authManager = FirebaseAuthManager.getInstance();
        dataSource = FirebaseDataSource.getInstance();
        userLiveData = new MutableLiveData<>();
        userLiveData.setValue(authManager.getCurrentUser());
    }

    public static synchronized AuthRepository getInstance() {
        if (instance == null) {
            instance = new AuthRepository();
        }
        return instance;
    }

    public LiveData<FirebaseUser> getUserLiveData() {
        return userLiveData;
    }

    public Task<AuthResult> login(String email, String password) {
        return authManager.getAuth().signInWithEmailAndPassword(email, password)
                .addOnSuccessListener(authResult -> userLiveData.setValue(authResult.getUser()));
    }

    public Task<AuthResult> register(String email, String password, String name) {
        return authManager.getAuth().createUserWithEmailAndPassword(email, password)
                .addOnSuccessListener(authResult -> {
                    FirebaseUser firebaseUser = authResult.getUser();
                    if (firebaseUser != null) {
                        // Create user profile in database
                        User user = new User(
                                firebaseUser.getUid(),
                                email,
                                name,
                                System.currentTimeMillis()
                        );
                        dataSource.getUserRef(firebaseUser.getUid()).setValue(user);
                        
                        // Send verification email
                        firebaseUser.sendEmailVerification();
                        
                        userLiveData.setValue(firebaseUser);
                    }
                });
    }

    public Task<Void> sendPasswordResetEmail(String email) {
        return authManager.getAuth().sendPasswordResetEmail(email);
    }

    public void logout() {
        authManager.signOut();
        userLiveData.setValue(null);
    }

    public boolean isUserLoggedIn() {
        return authManager.isUserLoggedIn();
    }

    public String getCurrentUserId() {
        return authManager.getCurrentUserId();
    }

    public FirebaseUser getCurrentUser() {
        return authManager.getCurrentUser();
    }
}

File: source/android/RiceDryer/app/src/main/java/com/qppd/ricedryer/data/repository/AuthRepository.kt
package com.qppd.ricedryer.data.repository

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.auth.UserProfileChangeRequest
import com.google.firebase.database.FirebaseDatabase
import com.qppd.ricedryer.data.model.UserProfile
import kotlinx.coroutines.tasks.await
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow

class AuthRepository {
    private val auth = FirebaseAuth.getInstance()
    private val database = FirebaseDatabase.getInstance()
    
    private val _currentUser = MutableStateFlow<FirebaseUser?>(auth.currentUser)
    val currentUser: Flow<FirebaseUser?> = _currentUser.asStateFlow()
    
    init {
        auth.addAuthStateListener { firebaseAuth ->
            _currentUser.value = firebaseAuth.currentUser
        }
    }
    
    suspend fun signUp(email: String, password: String, displayName: String): Result<FirebaseUser> {
        return try {
            val result = auth.createUserWithEmailAndPassword(email, password).await()
            val user = result.user ?: throw Exception("User creation failed")
            
            // Update profile with display name
            val profileUpdates = UserProfileChangeRequest.Builder()
                .setDisplayName(displayName)
                .build()
            user.updateProfile(profileUpdates).await()
            
            // Create user profile in database
            val userProfile = UserProfile(
                userId = user.uid,
                email = email,
                displayName = displayName,
                createdAt = System.currentTimeMillis(),
                lastLogin = System.currentTimeMillis()
            )
            
            database.getReference("users/${user.uid}")
                .setValue(userProfile)
                .await()
            
            Result.success(user)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
    
    suspend fun signIn(email: String, password: String): Result<FirebaseUser> {
        return try {
            val result = auth.signInWithEmailAndPassword(email, password).await()
            val user = result.user ?: throw Exception("Sign in failed")
            
            // Update last login
            database.getReference("users/${user.uid}/lastLogin")
                .setValue(System.currentTimeMillis())
                .await()
            
            Result.success(user)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
    
    suspend fun sendPasswordResetEmail(email: String): Result<Unit> {
        return try {
            auth.sendPasswordResetEmail(email).await()
            Result.success(Unit)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
    
    fun signOut() {
        auth.signOut()
    }
    
    fun getCurrentUser(): FirebaseUser? = auth.currentUser
    
    fun isUserSignedIn(): Boolean = auth.currentUser != null
}
